crucible/src/Lang/Crucible/Simulator/ExecutionTree.hs

```diff
@@ -194,7 +201,15 @@ gpGlobals = lens _gpGlobals (\s v -> s { _gpGlobals = v })
 -- TopFrame
 
 -- | The currently-executing frame plus the global state associated with it.
-type TopFrame sym ext f a = GlobalPair sym (SimFrame sym ext f a)
+--
+--   Type parameters:
+--
+--   - @sym@: instance of 'Lang.Crucible.Backend.IsSymInterface'
+--   - @ext@: language extension, see "Lang.Crucible.CFG.Extension"
+--   - @f@: the type of the top frame ('CrucibleLang' or 'OverrideLang')
+--   - @args@: arguments to this frame
```

Comment:
It might be worth explaining what the `Maybe` in `Maybe (Ctx CrucibleType)` indicates (and similarly for other uses of this type parameter).

Reply:
Done on `SimFrame`, linked to that from here (and elsewhere).